src/app/shell/admin-tools/platform/create-direction/directions/directions.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { Actions, ofAction, Select, Store } from '@ngxs/store';
+import { Observable, Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+import { ConfirmationModalWindowComponent } from 'src/app/shared/components/confirmation-modal-window/confirmation-modal-window.component';
+import { ModalConfirmationType } from 'src/app/shared/enum/modal-confirmation';
+import { NoResultsTitle } from 'src/app/shared/enum/no-results';
+import { Direction, DirectionsFilter } from 'src/app/shared/models/category.model';
+import { PaginationElement } from 'src/app/shared/models/paginationElement.model';
+import { DeleteDirectionById, FilterChange, FilterClear, GetFilteredDirections, PageChange, SetSearchQueryValue } from 'src/app/shared/store/admin.actions';
+import { AdminState } from 'src/app/shared/store/admin.state';
+
+@Component({
+  selector: 'app-directions',
+  templateUrl: './directions.component.html',
+  styleUrls: ['./directions.component.scss']
+})
+export class DirectionsComponent implements OnInit {
+
+  readonly noDirections = NoResultsTitle.noDirections;
+  searchValue = new FormControl('', [Validators.maxLength(200)]);
+  isResultPage = true;
+  searchedText: string;
+
+  @Select(AdminState.filteredDirections)
+  filteredDirections$: Observable<DirectionsFilter>;
+
+  @Select(AdminState.searchQuery)
+  searchQuery$: Observable<string>;
+
+  @Select(AdminState.isLoading)
+  isLoading$: Observable<boolean>;
+
+  destroy$: Subject<boolean> = new Subject<boolean>();
+
+  currentPage: PaginationElement = {
+    element: 1,
+    isActive: true
+  };
+
+  tabIndex: number;
+
+  constructor(
+    private store: Store,
+    private actions$: Actions,
+    private matDialog: MatDialog) { }
+
+
+  ngOnInit(): void {
+    this.store.dispatch([new FilterClear(), new GetFilteredDirections(), ]);
+
+      this.searchValue.valueChanges
+      .pipe(takeUntil(this.destroy$))
```

Comment:
add additional operators such as debounce time and startWith(''),  distinctUntilChanged(),
check it for the search-filed component/city filter, we have the same there